src/hotspot/share/runtime/thread.cpp

```diff
@@ -527,11 +527,6 @@ void Thread::print_owned_locks_on(outputStream* st) const {
 }
 #endif // ASSERT
 
-// We had to move these methods here, because vm threads get into ObjectSynchronizer::enter
-// However, there is a note in JavaThread::is_lock_owned() about the VM threads not being
-// used for compilation in the future. If that change is made, the need for these methods
-// should be revisited, and they should be removed if possible.
-
 bool Thread::is_lock_owned(address adr) const {
```

Comment:
Is there any reason not to move this to JavaThread now?  Also, I don't think it needs to be virtual.

Reply:
Good point. Only JavaThread's can own ObjectMonitors.

Reply:
OK yes - can move that to JavaThread, with just adding one cast in synchronizer.cpp, where 
ObjectSynchronizer::FastHashCode(Thread*, oop) uses is_lock_owned.

(ObjectSynchronizer::FastHashCode may be a candidate for taking JavaThread instead, maybe chasing down the users of that is a separate task. 8-) )

Reply:
The last time we talked about that we discovered that JVM/TI object tagging still created
fast hashcodes and that work is done by the VMThread (or some other non-JavaThread).